core/src/KokkosExp_MDRangePolicy.hpp

```diff
@@ -111,6 +112,22 @@ struct Rank {
   static constexpr Iterate inner_direction = InnerDir;
 };
 
+namespace Impl {
+// NOTE prefer C array U[M] to std::initalizer_list<U> so that the number of
+// elements can be deduced (https://stackoverflow.com/q/40241370)
+template <class Array, class U, std::size_t M>
+constexpr Array to_array_potentially_narrowing(const U (&init)[M]) {
+  using T = typename Array::value_type;
+  Array a{};
+  constexpr std::size_t N = a.size();
+  static_assert(M <= N, "");
+  auto* ptr = a.data();
+  for (auto x : init)
```

Comment:
Isn't this just

`std::transform(std::begin(init), std::end(init), [](U x){ return static_cast<T>(x); });`

or does that not work in a `constexpr` context?  If so, I prefer a comment as to why a raw loop.

I generally prefer algorithms, even the more obscure ones, both because they are self-documenting and less error-prone.

Reply:
I added a note.